Framework/Algorithms/src/PolarizationCorrections/HeliumAnalyserEfficiency.cpp

```diff
@@ -51,9 +51,9 @@ void HeliumAnalyserEfficiency::init() {
   auto validator = std::make_shared<CompositeValidator>();
```

Comment:
I don't think we're using this validator any more (it doesn't seem to work for WorkspaceGroup properties) so these three lines of code can be removed. We do, however, still need this validation that checks that the input workspace is in Wavelength units, as when I tested I was able to pass in a workspace in TOF units which shouldn't be possible. I would recommend taking a look at the workspace unit validation in either the `FlipperEfficiency` or `PolarizerEfficiency` (recently merged in) algorithms as we will need to add something similar here to ensure the validation takes place.

There is a unit test that checks for this, `testNonWavelengthInput`, however it's not checking the specific error message that's being thrown so I suspect there is some other reason why the algorithm is throwing an error here that is unrelated to the workspace units. I would recommend starting by re-working that test so that it's failing with the code changes here, and then look at adding the validation in to get the test to pass.